packages/tpetra/core/ext/TpetraExt_TripleMatrixMultiply_def.hpp

```diff
@@ -505,6 +504,7 @@ namespace Tpetra {
             targetMapToImportRow(i) = I_LID;
           }
         });
+      Kokkos::fence();
```

Comment:
I would have preferred putting the fences as close to the incorrect code as possible, with a giant comment saying "FIXME the code below is broken."  However, it's already great just to have these fences, and the Tpetra team can always grep for them and try to purge them later.